src/System.Private.CoreLib/shared/System/Globalization/ISOWeek.cs

```diff
@@ -0,0 +1,135 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using static System.DateTime;
+
+namespace System.Globalization
+{
+    public static class ISOWeek
+    {
+        private const int WeeksInLongYear = 53;
+        private const int WeeksInShortYear = 52;
+
+        private const int MinWeek = 1;
+        private const int MaxWeek = WeeksInLongYear;
+
+        public static int GetWeekOfYear(DateTime date)
+        {
+            int week = GetWeekNumber(date);
+
+            if (week < MinWeek)
+            {
+                return GetWeeksInYear(date.Year - 1);
+            }
+
+            if (week > GetWeeksInYear(date.Year))
+            {
+                return 1;
+            }
+
+            return week;
+        }
+
+        public static int GetYear(DateTime date)
+        {
+            int week = GetWeekNumber(date);
+
+            if (week < MinWeek)
+            {
+                return date.Year - 1;
+            }
+
+            if (week > GetWeeksInYear(date.Year))
+            {
+                return date.Year + 1;
+            }
+
+            return date.Year;
+        }
+
+        public static DateTime GetYearStart(int year)
+        {
+            return ToDateTime(year, MinWeek, DayOfWeek.Monday);
+        }
+
+        public static DateTime GetYearEnd(int year)
+        {
+            return ToDateTime(year, GetWeeksInYear(year), DayOfWeek.Sunday);
+        }
+
+        // From https://en.wikipedia.org/wiki/ISO_week_date#Weeks_per_year.
```

Comment:
>From https://en.wikipedia.org/wiki/ISO_week_date#Weeks_per_year. [](start = 11, length = 64)

would be good to add the comment:

The long years, with 53 weeks in them, can be described by any of the following equivalent definitions:

any year starting on Thursday (dominical letter D or DC) and any leap year starting on Wednesday (ED)
any year ending on Thursday (D, ED) and any leap year ending on Friday (DC)
years in which 1 January and 31 December (in common years) or either (in leap years) are Thursdays
All other week-numbering years are short years and have 52 weeks.